captum/optim/_utils/reducer.py

```diff
@@ -23,7 +23,11 @@ class ChannelReducer:
     """
 
     def __init__(
-        self, n_components: int = 3, reduction_alg: Any = "NMF", **kwargs
+        self,
+        n_components: int = 3,
+        reduction_alg: Any = "NMF",
+        supports_gpu: bool = False,

```

Comment:
The argument was for when the reducer algorithm supported GPU devices, as normally it converts inputs to CPU for the scikit-learn algorithm and then it puts the output back on the same device as the input. But I guess we could make so that the user has to put inputs on the CPU before running the class function.

Reply:
I'm removing this arg for now i.e. the user has the responsibility to put inputs on CPU before running the function if the reduction_alg is CPU-only. Also added a clearer error message if an error is raised because the user forgot to place it on CPU for the default sklearn NMF reducer.